GTG/plugins/gamify/gamify.py

```diff
@@ -0,0 +1,361 @@
+                self.data['streak'] += 1
+
+    def update_date(self):
+        today = date.today()
+        if self.data['last_task_date'] != today:
+            if self.data['last_task_number'] < self.preferences['goal']:
+                self.data['streak'] = 0
+            self.data['goal_achieved'] = False
+            self.data['last_task_number'] = 0
+            self.data['last_task_date'] = today
+
+    def get_current_level(self):
+        return min([(score, level) for score,level in self.LEVELS.items() if score >= self.get_score()])[1]
+
+    def get_score(self):
+        return self.data['score']
+
+    def get_number_of_tasks(self):
+        return self.data['last_task_number']
+
+    def get_streak(self):
+        return self.data['streak']
+
+    def button_update_score(self):
+        """Update the score in the BUTTON widget"""
+        score_label = self.builder.get_object('score_label')
+        score_label.set_markup(_("<b>{current_level}</b>").format(current_level=self.get_current_level()))
+        score_value = self.builder.get_object('score_value')
+        score_value.set_markup(_('You have: <b>{score}</b> points').format(score=self.get_score()))
+
+    def button_update_goal(self):
+        """Update the numbers of tasks done in the BUTTON widget"""
+        headerbar_label_button = self.builder.get_object('headerbar-label-button')
+        headerbar_label = self.builder.get_object('headerbar-label')
+        headerbar_msg = self.builder.get_object('headerbar-msg')
+
+        tasks_done = self.get_number_of_tasks()
+        goal = self.preferences['goal']
+        headerbar_label_button.set_markup("{tasks_done}/{goal}".format(tasks_done=tasks_done, goal=goal))
+
+        # Select a msg and emojo depending on the number of tasks done.
+        if tasks_done >= goal:
+            emoji = ["\U0001F60E", "\U0001F920", "\U0001F640"]
+            headerbar_label.set_markup(random.choice(emoji))
+            headerbar_msg.set_markup(_("Good Job!\nYou have achieved your goal."))
+        elif tasks_done >= 1:
+            emoji = ["\U0001F600", "\U0001F60C"]
+            headerbar_label.set_markup(random.choice(emoji))
+            headerbar_msg.set_markup(_("Only a few tasks to go!"))
+        else:
+            emoji = ["\U0001F643", "\U0001F648"]
+            headerbar_label.set_markup(random.choice(emoji))
+            headerbar_msg.set_markup(_("Get Down to Business\nYou haven't achieved any tasks today."))
+
+    def button_update_streak(self):
+        """Update the streak numbers in the BUTTON widget"""
+        streak_number = self.builder.get_object('streak_number')
+        streak_emoji = self.builder.get_object('streak_emoji')
+        if self.get_streak() > 0:
+            streak_emoji.set_markup("\U0001F525")
+        else:
+            streak_emoji.set_markup("\U0001F9CA")
+        streak_number.set_markup(_("You've completed your goal {streak} day in a row.").format(streak=self.get_streak()))
```

Comment:
You might need to use ngettext (if I recall correctly) to handle plurals for the streak here, for example day vs days... unless you cheat and consider it a streak only if `get_streak() > 1`  :)